### PR TITLE
APP-92 [Java] Basic level Apps: Credentials management

### PR DESCRIPTION
**Pull request by maxml**

### CODE REVIEW

File: credentialsdemo/source/java/src/org/kaaproject/kaa/examples/credentials/CredentialsDemo.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * A demo application that shows how to use the Kaa credentials API.
+ * z
+ *
+ * @author Maksym Liashenko
```
Comment (krupet) [inline credentialsdemo/source/java/src/org/kaaproject/kaa/examples/credentials/CredentialsDemo.java]: remove _author_  tag

File: credentialsdemo/source/java/src/org/kaaproject/kaa/examples/credentials/CredentialsDemo.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import org.kaaproject.kaa.client.profile.ProfileContainer;
+import org.kaaproject.kaa.schema.system.EmptyData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * A demo application that shows how to use the Kaa credentials API.
+ * z
```
Comment (krupet) [inline credentialsdemo/source/java/src/org/kaaproject/kaa/examples/credentials/CredentialsDemo.java]: Remove this.

File: credentialsdemo/source/java/src/org/kaaproject/kaa/examples/credentials/CredentialsDemo.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            @Override
+            public void onStarted() {
+                super.onStarted();
+                LOG.info("Kaa client started");
+
+                LOG.info("Endpoint ID:" + kaaClient.getEndpointKeyHash());
+
+            }
+        }, false);
+        kaaClient.setProfileContainer(new ProfileContainer() {
```
Comment (krupet) [inline credentialsdemo/source/java/src/org/kaaproject/kaa/examples/credentials/CredentialsDemo.java]: You don't need to set default profile container. Remove this part.

File: credentialsdemo/source/java/src/org/kaaproject/kaa/examples/credentials/CredentialsDemo.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    public static void main(String[] args) throws InterruptedException {
+/*
+         * Create the Kaa desktop context for the application.
+         */
+        DesktopKaaPlatformContext desktopKaaPlatformContext = new DesktopKaaPlatformContext();
+
+        /*
+         * Create a Kaa client and add a listener which displays the Kaa client
+         * configuration as soon as the Kaa client is started.
```
Comment (krupet) [inline credentialsdemo/source/java/src/org/kaaproject/kaa/examples/credentials/CredentialsDemo.java]: Update this comment. Does this app show configuration on startup?

File: credentialsdemo/source/java/src/org/kaaproject/kaa/examples/credentials/CredentialsDemo.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                return new EmptyData();
+            }
+        });
+
+        kaaClient.setFailoverStrategy(new CustomFailoverStrategy());
+        /*
+         * Persist configuration in a local storage to avoid downloading it each
+         * time the Kaa client is started.
+         */
+        kaaClient.setConfigurationStorage(new SimpleConfigurationStorage(desktopKaaPlatformContext, "saved_config.cfg"));
```
Comment (krupet) [inline credentialsdemo/source/java/src/org/kaaproject/kaa/examples/credentials/CredentialsDemo.java]: Does this app have some configuration? Remove this part.

File: credentialsdemo/source/java/src/org/kaaproject/kaa/examples/credentials/CredentialsDemo.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+    private static void readSymbol() {
+        try {
+            System.in.read();
+        } catch (IOException e) {
+            LOG.error("IOException has occurred: " + e.getMessage());
+        }
+    }
+
+    private static class CustomFailoverStrategy extends DefaultFailoverStrategy {
```
Comment (krupet) [inline credentialsdemo/source/java/src/org/kaaproject/kaa/examples/credentials/CredentialsDemo.java:89]: Provide some short description, why do we need this failover strategy and what it does?

File: credentialsdemo/src/main/java/org/kaaproject/kaa/examples/credentials/CredentialsDemoBuilder.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+import org.kaaproject.kaa.examples.common.AbstractDemoBuilder;
+import org.kaaproject.kaa.examples.common.KaaDemoBuilder;
+import org.kaaproject.kaa.server.common.admin.AdminClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * @author Maksym Liashenko
```
Comment (krupet) [inline credentialsdemo/src/main/java/org/kaaproject/kaa/examples/credentials/CredentialsDemoBuilder.java]: Remove _author_ tag.

File: credentialsdemo/resources/projects.xml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+             see 'sdkLanguage' type in demo_projects_schema.xsd schema) -->
+             
+        <sdkLanguage>JAVA</sdkLanguage>
+        
+        <!-- Specifies the target runtime platforms according to the instructions from the project details section
+             (ex. LINUX_X86, ANDROID, IOS, ESP8266, etc., to get list of all possible options
+             see 'platform' type in demo_projects_schema.xsd schema) -->
+             
+        <platforms>LINUX_X86 WINDOWS_X86</platforms>
+        <features>CONFIGURATION</features>
```
Comment (krupet) [inline credentialsdemo/resources/projects.xml:44]: I think we need to add one more option for _features_ in the common package. This must be replaced with something like  _DEVICE_MANAGEMENT_.

Comment (maxml) [inline credentialsdemo/resources/projects.xml:44]: Create task :+1:

**Comment by kaa-leeroy** [comment]
Can one of the admins verify this patch?

**Comment by Slash32** [comment]
@kaa-leeroy retest this please

**Comment by Slash32** [comment]
@kaa-leeroy retest this please

**Pull request merged**